### PR TITLE
Improve wifi-manager frontend

### LINKED ISSUE #513: core: frontend: Give users another chance... with wrong wifi passwords
Today, if the user try connecting on some wifi network with the wrong password, it will open an error notification.

A better approach would be to open a "trying to connect" state and get out of it when the request is answered, with fail or sucess. In case of fail because of wrong password, let the user try again.

DAYS_TO_DEVELOP:1

### LINKED ISSUE #533: frontend: Disconnect dialog spawns after successfully connecting to a network
DAYS_TO_DEVELOP: 0.3

### PR DESCRIPTION
**Pull request by rafaellehmkuhl**
This PR aims to improve (a lot) user's experience with wifi connection. Between the changes are:
- Add loading status while connecting
- Add connection result messages
- Raise wifi connection timeout, preventing cancelled requests
- Fix disconnection dialog appearing unexpectedly
- Reset password input box after connection
- Use backend's detailed error messages on connection

image
image
image

Needs #674

Fix #513
Fix #533 

Image available already.

### CODE REVIEW

File: core/frontend/src/components/wifi/ConnectionDialog.vue [hunk 1]
```diff
@@ -154,6 +179,18 @@ export default Vue.extend({
...
       return true
     },
+    connecting(): boolean {
+      return this.connection_status === ConnectionStatus.Connecting
+    },
+    tried_connecting(): boolean {
+      return [ConnectionStatus.Succeeded, ConnectionStatus.Failed].includes(this.connection_status)
+    },
+    connection_result_type(): string {
+      if (this.connection_status === ConnectionStatus.Succeeded) { return 'success' }
```
Comment (patrickelectric) [inline core/frontend/src/components/wifi/ConnectionDialog.vue:189]: A switch case would be more suitable

**Comment by patrickelectric** [comment]
Needs rebase

File: core/frontend/src/components/wifi/ConnectionDialog.vue (hunk 1 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/ConnectionDialog.vue:189]: Wait a second and I will add this.

File: core/frontend/src/components/wifi/ConnectionDialog.vue [hunk 2]
```diff
@@ -164,24 +215,28 @@ export default Vue.extend({
...
+          this.connection_result_message = `Successfully connected to '${this.real_ssid}'`
           wifi.setNetworkStatus(null)
+          this.password = ''
+          this.force_password = false
         })
         .catch((error) => {
-          const message = `Could not connect to wifi network: ${error.message}.`
+          this.connection_status = ConnectionStatus.Failed
+          const message = error.response.data.detail ? error.response.data.detail : `Connection failed: ${error}`
+          this.connection_result_message = message
```
Comment (patrickelectric) [inline core/frontend/src/components/wifi/ConnectionDialog.vue:237]: We could add a message saying to check if the wifi password is correct.

Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/ConnectionDialog.vue:237]: Definitely. Adding.

**Pull request merged**